doc/effects.md

```diff
@@ -1,268 +1,428 @@
-[here](https://github.com/flame-engine/flame/tree/main/examples/lib/stories/effects/opacity_effect.dart).
+- **`duration`** -- the length of the main part of the effect, i.e. how long it should take to go
+  from 0 to 100%. This parameter cannot be negative, but can be zero. If this is the only parameter
+  specified then the effect will grow linearly over the `duration` seconds.
+
+- **`curve`** -- if given, creates a non-linear effect that grows from 0 to 100% according to the
+  provided [curve](https://api.flutter.dev/flutter/animation/Curves-class.html).
+
+- **`reverseDuration`** -- if provided, adds an additional step to the controller: after the effect
+  has had grown from 0 to 100% over the `duration` seconds, it will then grow back from 100% to 0
+  over the `reverseDuration` seconds. In addition, the effect will complete at progress level of 0
+  (normally the effect completes at progress 1).
+
+- **`reverseCurve`** -- the curve to be used during the "reverse" step of the effect. If not given,
+  this will default to `curve.flipped`.
+
+- **`alternate`** -- setting this to true is equivalent to specifying the `reverseDuration` equal
+  to the `duration`. If the `reverseDuration` is already set, this flag has no effect.
+
+- **`atMaxDuration`** -- if non-zero, this inserts a pause after the effect reaches its max
+  progress and before the reverse stage. During this time the effect is kept at 100% progress. If
+  there is no reverse stage, then this will simply be a pause before the effect is marked as
+  completed.
 
-## ColorEffect
+- **`atMinDuration`** -- if non-zero, this inserts a pause after the reaches its lowest progress
+  (0) at the end of the reverse stage. During this time, the effect's progress is at 0%. If there
+  is no reverse stage, then this pause will still be inserted after the "at-max" pause if it's
+  present, or after the forward stage otherwise. In addition, the effect will now complete at
+  progress level of 0.
 
-This effect will change the base color of the paint, causing the rendered component to be tinted by
-the provided color.
+- **`repeatCount`** -- if greater than one, it will cause the effect to repeat itself the prescribed
+  number of times. Each iteration will consists of the forward stage, pause at max, reverse stage,
+  then pause at min (skipping those that were not specified).
 
-Usage example:
+- **`infinite`** -- if true, the effect will repeat infinitely and never reach completion. This is
+  equivalent to as if `repeatCount` was set to infinity.
+  
+- **`startDelay`** -- an additional wait time inserted before the beginning of the effect. This
+  wait time is executed only once, even if the effect is repeating. During this time the effect's
+  `.started` property returns false. The effect's `onStart()` callback will be executed at the end
+  of this waiting period.
+
+  Using this parameter is the simplest way to create a chain of effects that execute one after
```

Comment:
Isn't it simpler to chain them in `onComplete`, since you don't have to think about the timings then? At least when the effects aren't alternating, depending on how you want your chain to act.

Reply:
It might be a matter of preference. For me, adding several effects each with a different delay sounds like the simplest way to chain them (other than a `SequenceEffect`). Adding new effects in the `onComplete` callback is an alternative solution, that sounds more complicated to me (the code becomes nested deeper), but on the other hand it avoids arithmetics with the durations. Using the delays has an additional advantage that you can make the effects overlap if necessary, which is not possible with other solutions.

Given that we still haven't implemented `SequenceEffect`, or the `onComplete` as an explicit parameter, I'd say we can leave this sentence as-is.